### PR TITLE
Release constant placement

### PR DESCRIPTION
**Pull request by tanyokwok**
The `OpsPlacer` pass will place i32 `mhlo::ConstOp` on the CPU because it’s considered to be used to calculate shapes heuristically. But, shapes will be only 1D-Tensors(vectors) or constructed from Scalars.

The PR releases the `mhlo::ConstOp` with ranks greater than 1 from placing to CPU in the `OpsPlacer`  pass.

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/mhlo_placer.cc [hunk 1]
```diff
@@ -461,7 +465,7 @@ void OpsPlacer::placeI32Ops() {
             op->getOperand(i).getType().dyn_cast<RankedTensorType>();
         if (!operand_ty) continue;
         auto elem_type = operand_ty.getElementType();
-        if (elem_type.isInteger(32)) {
+        if (elem_type.isInteger(32) && operand_ty.getRank() <= 1) {
```
Comment (wyzero) [inline tao_compiler/mlir/disc/transforms/mhlo_placer.cc:468]: I think check the number of elements maybe more reliable?

**Comment by tanyokwok** [comment]
mark related to https://github.com/alibaba/BladeDISC/issues/766

**Comment by qiuxiafei** [comment]
Any updates?

**Comment by tanyokwok** [comment]
> Any updates?

No updates, I will try to resolve the issue by canonicalization.

**Comment by tanyokwok** [comment]
The PR must be outdated

**Pull request closed without merging**